Infrastructure/Resources/New-ResourceGroup.ps1

```diff
@@ -12,36 +12,114 @@ The location of the Resource Group. This is limited to West Europe and North Eur
 .PARAMETER Name
 The name of the Resource Group
 
+.PARAMETER TagConfigPath
+Path to the Tag configuration JSON file .contents example
+
+{
+    "Name": "CP",
+    "Value": {
+        "BSA": "DAS",
+        "SA": "DAS",
+        "ENV": "PD",
+        "WL": "TBD",
+        "DC": "TBD",
+        "BC": "DAS"
+    }
+}
+
+.EXAMPLE
+
+.\New-ResourceGroup.ps1 -Name arm-rg-01 -Location "West Europe" 
+
 .EXAMPLE
-.\New-ResourceGroup.ps1 -Name arm-rg-01 -Location "West Europe"
+
+.\New-ResourceGroup.ps1 -Name arm-rg-01 -Location "West Europe" -TagConfigPath "c:\temp\TagConfig.json"
 
 #>
 
 Param (
-    [Parameter(Mandatory = $true)]
+    [Parameter(Mandatory = $True)]
     [ValidateSet("West Europe", "North Europe")]
     [String]$Location,
-    [Parameter(Mandatory = $true)]
+    [Parameter(Mandatory = $True)]
+    [ValidateNotNullOrEmpty()]
+    [String]$Name,
+    [Parameter(Mandatory = $False)]
     [ValidateNotNullOrEmpty()]
-    [String]$Name
+    [String]$TagConfigPath
 )
 
 # --- Import Azure Helpers
 Import-Module (Resolve-Path -Path $PSScriptRoot\..\Modules\Azure.psm1).Path
 Import-Module (Resolve-Path -Path $PSScriptRoot\..\Modules\Helpers.psm1).Path
 
-Write-Log -LogLevel Information -Message "Checking for existing Resource Group $Name"			 
+Write-Log -LogLevel Information -Message "Checking for existing Resource Group $Name"
 $ExistingResourceGroup = Get-AzureRmResourceGroup -Name $Name -ErrorAction SilentlyContinue
 
+# --- Enumerate Tag Value's from Config File if Parameter is set 
+If ($TagConfigPath) {
+    $TagConfig = Get-Content -Path $TagConfigPath -Raw | ConvertFrom-Json
+    $ConfigTagValue = $TagConfig.Value | ConvertTo-Json
+    $ConfigTagName = $TagConfig.Name 
+    # --- Define & Set variables for Tags
+    $TagPresentValueCorrect = $True 
+    $TagValueUpdated = $False
+    $Tags = $ExistingResourceGroup.Tags
```

Comment:
$TagPresentValueCorrect, $TagValueUpdated and $Tags are not used until later on and seem out of place in this block.

Reply:
This is taken out of context from the IF statement . I believe it needs to be in this block